### PR TITLE
update node to v20 lts

### PR DESCRIPTION
**Pull request by akellbl4**
- updates node to v20
- enable use of pnpm v8
- replaces nvmrc with npmrc ([pnpm manages node version for you](https://pnpm.io/npmrc#use-node-version))
- moves env requirements to the root `package.json`

### CODE REVIEW

File: frontend/packages/api/lib/fetcher.ts [hunk 1]
```diff
@@ -79,6 +79,7 @@ export const createFetcher = (site: string, baseUrl: string): Client => {
 				.catch(Object)
 				.then((data: string) => {
 					if (res.status < 200 || res.status > 299) {
+						console.log(data)
```
Comment (Mavrin) [inline frontend/packages/api/lib/fetcher.ts:82]: please remove console.log

File: frontend/packages/api/tests/clients/admin.test.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ interface Context {
 	client: ReturnType<typeof createAdminClient>
 }
 
-describe<Context>('Admin Client', (adminClient) => {
+describe.skip<Context>('Admin Client', (adminClient) => {
```
Comment (Mavrin) [inline frontend/packages/api/tests/clients/admin.test.ts:9]: why do you skip this test

File: frontend/e2e/screenshot-firefox.png
Comment (Mavrin) [inline frontend/e2e/screenshot-firefox.png:1]: why does firefox show only one comment? I see that other browsers render 5 comments.

**Review by Mavrin — CHANGES_REQUESTED** [review]

File: frontend/e2e/screenshot-firefox.png
Comment (akellbl4) [reply frontend/e2e/screenshot-firefox.png:1]: All of the tests started to fail after update, so I skip them. This code is not in use so I wanted to fix it as follow up

File: frontend/packages/api/tests/clients/admin.test.ts (hunk 2 above)
Comment (akellbl4) [reply frontend/packages/api/tests/clients/admin.test.ts]: reworked the whole thing, tests are fixed.

File: frontend/e2e/screenshot-firefox.png
Comment (akellbl4) [reply frontend/e2e/screenshot-firefox.png:1]: Ops, those shouldn't get into the repo.
Those were created while local testing and FF had some problems.

File: frontend/packages/api/package.json [hunk 3]
```diff
@@ -1,57 +1,68 @@
...
   "author": "Paul Mineev",
   "license": "MIT",
+  "type": "module",
+  "main": "dist/index.js",
+  "types": "dist/index.d.ts",
+  "exports": {
+    ".": {
+      "import": "./dist/index.js",
+      "require": "./dist/index.js",
+      "types": "./dist/index.d.ts"
```
Comment (Mavrin) [inline frontend/packages/api/package.json:21]: https://nodejs.org/api/packages.html#community-conditions-definitions
> "types" - can be used by typing systems to resolve the typing file for the given export. This condition should always be included first.

File: frontend/packages/api/src/lib/fetcher.ts [hunk 4]
```diff
@@ -90,5 +87,23 @@ export const createFetcher = (site: string, baseUrl: string): Client => {
...
+
+function getSearchParams(query?: QueryParams) {
+	if (!query) {
+		return ''
+	}
+
+	// overrides type of init in URLSearchParams constructor because it's not correct and accepts QueryParams
+	const searchParams = new URLSearchParams(
+		Object.entries(query).reduce<Record<string, string>>(
+			(acc, [k, v]) => Object.assign(acc, { [k]: v.toString() }),
```
Comment (Mavrin) [inline frontend/packages/api/src/lib/fetcher.ts:101]: I think we don't need to create new object on every iterations and we can mutate `acc`
```
acc[k] = v.toString();
```

**Review by Mavrin — CHANGES_REQUESTED** [review]

**Pull request closed without merging**